### PR TITLE
Fetching USB speed and Myriad X serial number

### PR DESCRIPTION
**Pull request by saching13**
Modified `initFromHostSide` to take `usb_speed` and `mx_serial` strings as arguments(pass by reference) which are private variables of device class. allowing them to be loaded on XLink connection.

### CODE REVIEW

File: src/xlink/xlink_wrapper.cpp [hunk 1]
```diff
@@ -212,7 +213,8 @@ bool XLinkWrapper::initFromHostSide(
     _reboot_device_on_destructor = reboot_device_on_destructor;
 
     assert(_device_link_id == -1);
-
+    // variable to copy mriad x serial id and usb speed
+    char dev_conn_info[148] = { 0 };
```
Comment (szabi-luxonis) [inline src/xlink/xlink_wrapper.cpp:217]: this is unused?

File: src/xlink/xlink_wrapper.cpp [hunk 2]
```diff
@@ -295,7 +296,7 @@ bool XLinkWrapper::initFromHostSide(
             // Development option, the firmware is loaded via JTAG
             printf("Device boot is skipped. (\"binary to boot from\" NOT SPECIFIED !)\n");
         }
-
+        printf("firmware sent\n");
```
Comment (szabi-luxonis) [inline src/xlink/xlink_wrapper.cpp:299]: I assume this is just debug print?

File: src/xlink/xlink_wrapper.cpp [hunk 3]
```diff
@@ -341,6 +342,23 @@ bool XLinkWrapper::initFromHostSide(
...
+    std::vector<std::string> speed_str = {"Unknown", "Low/1.5Mbps", "Full/12Mbps", "High/480Mbps", "Super/5000Mbps"};
+    return speed_str[XLinkGetUSBSpeed(_device_link_id)];
+}
+
+std::string XLinkWrapper::getMxSerial(){
+    assert(_device_link_id != -1);
+    return std::string(XLinkGetMxSerial(_device_link_id)); 
+}
+
+XLinkError_t XLinkWrapper::connect(XLinkHandler_t* handler){
```
Comment (szabi-luxonis) [inline src/xlink/xlink_wrapper.cpp:357]: can you rename it to `XLinkConnectSafe` or something similar, so it's clear that it's just a wrapper ?

**Review by szabi-luxonis — CHANGES_REQUESTED** [review]

File: src/xlink/xlink_wrapper.cpp (hunk 2 above)
Comment (saching13) [reply src/xlink/xlink_wrapper.cpp:299]: Yes. yes. I haven't cleaned this one yet. let me clean it up. :)

**Review by szabi-luxonis — APPROVED** [review]
LGTM

File: src/xlink/xlink_wrapper.cpp [hunk 4]
```diff
@@ -341,6 +339,23 @@ bool XLinkWrapper::initFromHostSide(
 
     return result;
 }
+
+std::string XLinkWrapper::getUSBSpeed(){
```
Comment (themarpe) [inline src/xlink/xlink_wrapper.cpp:343]: Return UsbSpeed_t here, and create a separate function which outputs usb speed as string (if needed).
And modify this: https://github.com/luxonis/depthai-core/pull/14/files#r517970300

**Pull request merged**